File: src/g1ipmlinkedlist/Go/llist.go
package main

import "fmt"

// LinkedList represents the linked list and includes methods for manipulation.
type LinkedList struct {
	head *LinkedListNode
}

// InsertNodeAtHead inserts a node at the head of the linked list.
func (l *LinkedList) InsertNodeAtHead(node *LinkedListNode) {
	if l.head == nil {
		l.head = node
	} else {
		node.next = l.head
		l.head = node
	}
}

// CreateLinkedList creates a linked list from a slice of integers.
func (l *LinkedList) CreateLinkedList(lst []int) {
	for i := len(lst) - 1; i >= 0; i-- {
		newNode := InitLinkedListNode(lst[i])

		l.InsertNodeAtHead(newNode)
	}
}

// DisplayLinkedList displays the linked list elements as an array.
func (l *LinkedList) DisplayLinkedList() {
	temp := l.head

	fmt.Print("[")
	for temp != nil {
		fmt.Print(temp.data)
		temp = temp.next
		if temp != nil {
			fmt.Print(", ")
		}
	}
	fmt.Print("]")
}

// reverse reverses the linked list in place.
func reverse(head *LinkedListNode) *LinkedListNode {
	var prev, next *LinkedListNode
	current := head

	for current != nil {
		next = current.next
		current.next = prev
		prev = current
		current = next
	}

	return prev
}

File: src/g1ipmlinkedlist/Python/llist.py
from llist_node import init_linked_list_node


class LinkedList:
    def __init__(self):
        self.head = None

    def insert_node_at_head(self, node):
        if self.head is None:
            self.head = node
        else:
            node.next = self.head
            self.head = node

    def create_linked_list(self, lst):
        for i in reversed(lst):
            new_node = init_linked_list_node(i)
            self.insert_node_at_head(new_node)

    def display_linked_list(self):
        temp = self.head
        elements = []
        while temp is not None:
            elements.append(temp.data)
            temp = temp.next
        print(elements)


def reverse(head):
    prev = None
    current = head
    while current is not None:
        next_node = current.next
        current.next = prev
        prev = current
        current = next_node
    return prev
